code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/PropositionsRepository.kt

```diff
@@ -0,0 +1,17 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.optimize
+
+interface PropositionsRepository {
```

Comment:
Why do we need this interface? `PropositionsRepository` is a singleton class. We are never expected to have multiple classes implementing this interface. I'm not sure what it achieves

Reply:
removed @spoorthipujariadobe, it made sense when we're maintaining a list of propositions